### ISSUE TITLE
Use available RAM instead of the hard disk (`/var/tmp`) when feasible, for increased performance and to avoid filling up the storage partitions

### ISSUE THREAD

**Issue by nekohayo** [audacity/audacity#2819]
**To Reproduce**
1. Drag & drop a 200 MB (3 hours long) opus audio file (a MP3 would probably work too) into Audacity when the disk holding `/var/tmp` has less than 5 GB of available space

**Actual behavior**
Audacity is very slow at opening the file because it has to write it all out uncompressed to disk in `/var/tmp`, and eventually fills up that disk 100% and will choke itself. In the case above, it tried opening that 3-hours-long audio file, and at merely 30% of the open/import process I noticed that my hard disk was full; it had consumed 2.4 GB trying to decompress that file, without telling me.

What's worse is that if run in a sandboxed environment like flatpak, /var/tmp is not truly on /var/tmp, but secretly in the user's home dir inside the sandbox... and when you fill a home directory to 100% (like would be the case here because my home dirs are always super tight on disk space; please don't ask me to buy new drives, a natural law of this world is that one week afterwards they will be full again), Very Bad Things happen. Programs will corrupt their user data, crash, etc. So you shouldn't fill that directory without the user knowing about it.

**Expected behavior**
Audacity knows how long the file is (in duration etc.), how much RAM I have, and how much disk space I have available for its temp dir.

In the case at hand, I have way more RAM than available disk space; with my machine with 16 GB of RAM (let alone my other machine with 24 GB of RAM) it could easily (and nearly-instantly) have opened/imported that 3-hours-long audio file solely in RAM. It would have been blazing fast, saved my SSD from pointless wear & tear, and it would have avoided filling up to the point where I only had 100 MB left and urgently tried to kill the Audacity process (but nearly failed because the machine was getting so hammered by disk I/O that it didn't want to respond quickly and the app didn't want to be killed, it became a zombie process instead of being killed)

I shouldn't have to know this, nor have to assign a dedicated 20 GB drive just for Audacity for the 1-2 times a year I'm attempting to open a 3-hours-long audio file. To be clear, I'm not asking for a "setting" here, I shouldn't have to configure this behavior in the Preferences at all: software is supposed to be smarter than humans, and the parameters are known/calculatable, so I believe Audacity should automatically figure this stuff out for me.

If opening large files and I have lots of RAM, or if I have more RAM than free disk space, open it in RAM, don't waste time trying the disk. Or at _least_ warn me ahead of time that the operation cannot happen because of insufficient space for the corresponding file duration, but preferrably Audacity should automatically figure out the solution before and instead.

**Additional information (please complete the following information):**
 - OS: Fedora Linux 35
 - Version: Audacity 3.1.3, in a Flatpak from Flathub

**Comment by LWinterberg** [comment]
duplicate of #614 

regardless, I'm not entirely sure whether this behavior would be preferable:

3 hours of uncompressed 32 bit 48 kHz stereo would be about 4 GB, if you were to apply 4 effects (say: noise reduction, EQ, compressor, normalization) across it, you'd already be at 16 GB and crash your system by running out of memory due to having undo states - and if instead of just importing a file you would have made a recording, you'd lose the entire recording as well

**Comment by nekohayo** [comment]
Recording is kind of a very different thing from "open an existing audio file" in my view as a user, but I see what you mean. I just don't think users should have to care about this merely for opening files, and that in an ideal world (I know, that's a big if, it's easier said than done, and I realize I'm asking for a lot here) operations/effects/etc. should be non-destructive or using diffs instead of replicating the whole uncompressed timeline length each time (this is just a wild uneducated guess, I have no idea how Audacity is architected in practice); I'd be quite surprised if you told me other apps still use this same "every action in the history is an entire snapshot of everything, uncompressed" approach nowadays...

**Comment by LWinterberg** [comment]
I think the big difference is that the only file type larger than audio (video) is always non-destructive, so the undo states are just tiny instructions, and anything else (images, text) is so tiny that you never run into this problem one way or another. 

The Adobe suite will decompress any media files into a media & cache folder (which is capped to 250 GB I believe) and use RAM for non-destructive undo steps, as well as the pre-render cache. And in order for you to not lose progress on crash, it creates a bunch of backup project files for you within this 250GB limit every few minutes. 

This does mean however that their system fails exactly as bad as Audacity when their `/var/tmp` equivalent is limited to 5 GB (or 1GB, as was the case on a university device I used). In that case, you'd need to specify a different cache location before starting to work - just like you'd do in Audacity's Directories Preferences.

**Comment by petersampsonaudacity** [comment]
Older versions of Audacity did have this facility (see the example for 1.3.14 below).  

a) Note carefully that is states "useful for slow drives".  In the days of early Audacity many drives were slow but these days this is (mostly) no longer the case.  
b) And @LWinterberg is right about the running out of space issue in RAM.

Which is way this option was removed from Audacity a while back.

![image](https://user-images.githubusercontent.com/50205984/163385872-911efed3-6e7d-43bc-8422-a5ee36333bd0.png)

**Comment by petersampsonaudacity** [comment]
@nekohayo   have you thought of trying to set up a virtual disk in RAM and then use that as a location for
a) Audacity temporary files (i.e. unsaved projects)
b) Saving a project (empty) and then importing into it

I'd be fascinated to know if that works.
